WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -13,40 +13,27 @@ struct ItemListView: View {
     @Binding var selectedItemListType: ItemListType
     @Binding var searchTerm: String
 
-    private var shouldShowSearchField: Bool {
-        selectedItemListType == .products(search: true)
-    }
     @FocusState private var isSearchFieldFocused: Bool
 
     @State private var searchTask: Task<Void, Never>?
     @State private var didFinishSearch = true
 
-    var itemsController: PointOfSaleItemsControllerProtocol {
-        switch selectedItemListType {
-        case .products(search: false):
-            posModel.purchasableItemsController
-        case .products(search: true):
-            posModel.purchasableItemsSearchController
-        case .coupons:
-            posModel.couponsController
-        }
-    }
-
-    private var itemListState: ItemListState {
-        itemsController.itemsViewState.itemsStack.root
+    private var isCouponsFeatureEnabled: Bool {
+        ServiceLocator.featureFlagService.isFeatureFlagEnabled(.enableCouponsInPointOfSale)
     }
 
-    private var shouldShowCoupons: Bool {
-        ServiceLocator.featureFlagService.isFeatureFlagEnabled(.enableCouponsInPointOfSale)
+    private var isSearchProductsFeatureEnabled: Bool {
+        ServiceLocator.featureFlagService.isFeatureFlagEnabled(.searchProductsInPOS)
     }
 
     private var isAddingCouponAllowed: Bool {
         guard case .coupons = selectedItemListType else { return false }
+        let itemListState = itemListState(selectedItemListType)
```

Comment:
I can't find any problems with this state indirection, but I have the fear! I've tested things like variations, adding a coupon, PTR, load more, empty state, it all seems fine.

Reply:
I don't think you should be fearful. In this case, the behavior is identical; we take an itemListState of `selectedItemListType`.